### PR TITLE
Create giveRoleAfterDelay

### PR DESCRIPTION
**Pull request by JordiHakimZiyech**
gives role after delay cuz why not huh

### CODE REVIEW

File: util/giveRoleAfterDelay [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ 
+	Recommended trigger: timedrole
+	Trigger type: Command
+ 
+	
+*/}}
+{{/*ACTUAL CODE, DO NOT TOUCH*/}}
+{{ if not .ExecData }}
+{{ $args := parseArgs 3 "Wrong usage: `-timedrole (@user) (time) (role name)`" 
+ (carg "userid" "member to give role" )
```
Comment (Jo3-L) [inline util/giveRoleAfterDelay:13]: `member` carg and access `User` field. `userid` doesn't validate.

File: util/giveRoleAfterDelay [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ (carg "duration" "time till role" )
+ (carg "string" "role name" ) }}
+{{ $search := slice .CmdArgs 2 | joinStr "" | lower }}
```
Comment (Jo3-L) [inline util/giveRoleAfterDelay:16]: choose one and be consistent, `parseArgs` or manually parsing

File: util/giveRoleAfterDelay [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ {{- range .Guild.Roles -}}
+ {{ if not $role }}
+ {{ $rName := lower .Name }}
+ {{ if or (eq $search $rName) (in $rName $search) }}
+ {{ $role = . }}
+ {{ end }}
+ {{ end }}
+ {{- end -}}
+{{ if $role }}
+{{ $a := (toDuration ($args.Get 1)).Seconds }}
```
Comment (Jo3-L) [inline util/giveRoleAfterDelay:28]: no need, already a duration.

File: util/giveRoleAfterDelay [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+{{ if not (hasRoleName ($args.Get 2)) }}
+{{ execCC .CCID nil $a (sdict "user" ($args.Get 0) "role" ($args.Get 2)) }}
```
Comment (Jo3-L) [inline util/giveRoleAfterDelay:30]: this won't work properly with fuzzy matching. Say user provides `foo` as their input role and the role that matches is `foobar`. `foo` is not the role name, `foobar` is, but you're using `foo`. Use role ID instead.

File: util/giveRoleAfterDelay [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+the role: {{ $role.Name }} will be given in {{$args.Get 1}} 
+{{ else }} The user has this role already {{ end }} 
+{{ else }} Role not found, make sure its correct {{ end }}
+{{ else }} {{ $user := .ExecData.user }}
+{{ $role := .ExecData.role }}
+{{ giveRoleName $user $role }}
+role has been given to <@{{$user}}> {{end}}
```
Comment (Jo3-L) [inline util/giveRoleAfterDelay:37]: dunno if message should be sent after delay, especially if it's long and people forget

**Review by Jo3-L — CHANGES_REQUESTED** [review]
maybe support cancelling giving the role and override past invocations using `scheduleUniqueCC` rather than `execCC`.

**Comment by JordiHakimZiyech** [comment]
FRICK FILE NAME, SHIT

**Pull request closed without merging**